docs/sources/style-guide/write-for-developers/index.md

```diff
@@ -0,0 +1,148 @@
+There is no need to explain elementary ideas. Instead, introduce those concepts and features that are specific to Grafana Labs products.
+For example, instead of covering the fundamentals of UI design  _in general_, explain how Grafana Labs software or APIs _interpret_ those principles.
+
+## Code comments
+
+The foundation of strong documentation is well-written comments in code that are concise, relevant, and current. 
+
+For do's and don'ts of writing comments, refer to the [Guidelines for code comments in grafana-* packages](https://github.com/grafana/grafana/blob/main/contribute/style-guides/code-comments.md).
+
+For more general advice, consult one of the reputable [Google Style Guides](https://google.github.io/styleguide/) for your favorite programming language. 
+
+## Reference docs
+
+Whenever possible, automatically generate API and other reference documentation from source-code comments. However documentation is created, make sure it conforms to the [style conventions]({{< relref "../../style-guide/style-conventions/" >}}).
+Pay particular attention to properly formatting the elements of code.
+
+### Auto-generated documentation
+
+The advantages of automating documentation programmatically are well-known, and include increased consistency and a reduction in human-made errors.
+But behind every line of auto-generated content is a human author who is responsible for following the [style guide]({{< relref "../../style-guide/" >}}).
+
+When writing documentation that will be used by an auto-generated program to create publishable content, keep the following things in mind:
+
+An auto-generation tool can parse syntax, but when writing documentation that will be used to create automated content, it is up to you to add actionable insights. 
+- What are the caveats, edge cases, and side effects?
+- What is the bigger picture that is not self-evident in the code? 
+- In short, what is everything that a developer needs to know to use the code?
+
+> **Note:** It might be difficult to integrate auto-generated content with other documentation, such as relevant sections of _Get started_ guides, tutorials, or detailed code examples. You might need to ask a Docs team member to ensure that your content is properly cross-referenced.
+
+### Elements of API references
+
+Properly document the most common elements of an API reference, such as the title, parameters, return values, and so on. The following suggestions will help you to write more complete and consistent documentation:
+
+| Element                                               | Description                                                           |
+| ------------------------------------------------- | ----------------------------------------------------------------------- |
+| Title and description       | The name of the element and a description of it in one or two sentences. Put API names, classes, methods, etc. in `code font`. |
```

Comment:
```suggestion
| Title and description       | The name of the element and a description of it in one or two sentences. Put API names, classes, methods, and so forth in `monospace`. |
```

Reply:
Note: we could say `code font`, but I've used `monospace` most places. I won't edit that nit further, based on what you decide.

Reply:
I would argue for going with `code font` because technically it is produced with a backtick in Markdown, and backticks are semantic markers for code, not for a monospace style.

Reply:
I have usually called this `fixed-width` font.

Reply:
changed to "fixed-width font" for now.